glotaran/parse/register.py

```diff
@@ -1,7 +1,12 @@
 """A register for models"""
-import typing
+from __future__ import annotations
 
-from glotaran.model.model import Model
+from typing import TYPE_CHECKING
+
+if TYPE_CHECKING:
+    from typing import List
+
+    from glotaran.model import Model
```

Comment:
Looks a bit odd, but I guess if it's a standard pattern, just go for it.

Reply:
Since the import is only used for type checking, it kinda makes sense to only import it when actually type checking.